psa-crypto-sys/src/constants.rs

```diff
@@ -25,6 +25,9 @@ pub const PSA_ERROR_HARDWARE_FAILURE: psa_status_t = -147;
 pub const PSA_ERROR_INSUFFICIENT_ENTROPY: psa_status_t = -148;
 pub const PSA_ERROR_INVALID_SIGNATURE: psa_status_t = -149;
 pub const PSA_ERROR_INVALID_PADDING: psa_status_t = -150;
+pub const PSA_ERROR_CORRUPTION_DETECTED: psa_status_t = -151;
+pub const PSA_ERROR_DATA_CORRUPT: psa_status_t = -152;
+pub const PSA_ERROR_DATA_INVALID: psa_status_t = -153;
 pub const PSA_ERROR_INSUFFICIENT_DATA: psa_status_t = -143;
```

Comment:
The last two (-143 and -136) seem to be in the wrong place but that's not part of this patch so I shouldn't really complain.

Reply:
Remind me why these values can't come from the C library somehow? In a previous patch we had something like `pub const PSA_SOMETHING: psa_something_t = shim_PSA_SOMETHING;` in the Rust code and `const psa_something_t shim_PSA_SOMETHING = PSA_SOMETHING;` in `shim.h`. It would be nice to explain in a comment at the top of `constants.rs` why these values are being repeated like this.

Reply:
We removed from the shim all the values that are defined by the specification. That way, it allows you to use those value, without having to link with a PSA Crypto implementation (that is the `implementation-defined` feature). That is useful for Parsec interface where we re-use some of these values but do not want to link to a PSA Crypto implementation.

Reply:
At least I hope I removed all the duplication, as I did everything manually

Reply:
I hadn't realised some values are defined by the specification. Should that be mentioned in a comment in `constants.rs`?

Reply:
I think that makes sense - we do have a module comment in this file, it might be worth expanding that to explain the logic of what's hidden by `implementation-defined` - there's a brief explanation [here](https://github.com/parallaxsecond/rust-psa-crypto/tree/master/psa-crypto-sys) but it should really be in docs.rs too.